synapse/config/workers.py

```diff
@@ -118,12 +138,41 @@ def read_config(self, config, **kwargs):
                 )
             )
 
-        # Whether to send federation traffic out in this process. This only
-        # applies to some federation traffic, and so shouldn't be used to
-        # "disable" federation
-        self.send_federation = config.get("send_federation", True)
+        # Handle federation sender configuration.
```

Comment:
It looks like the logic for sharding federation sender and push is identical, can we abstract this somehow?

I think something like `self.send_federation = self._handle_shared_config("send_federation", "federation_sender_instances", "synapse.app.federation_sender")` might do it?

Reply:
I'm slightly in two minds about this. On the one had, yes, on the other we should never have more than these two, and so I think it might actually be clearer just writing it out rather than factoring it out and passing a bunch of stuff in etc :shrug:

Reply:
If there isn't additional workers that we'll need to apply this to than that sounds OK.

Reply:
Yeah, this only for working around the old style config options, new stuff shouldn't have that problem